code/game/gamemodes/objective.dm

```diff
@@ -384,8 +384,18 @@ var/list/potential_theft_objectives = subtypesof(/datum/theft_objective) \
 	return steal_target
 
 /datum/objective/steal/check_completion()
-	if(!steal_target) return 1 // Free Objective
-	return steal_target.check_completion(owner)
+	if(!steal_target)
+		return 1 // Free Objective
+
+	var/list/all_items = owner.current.GetAllContents()
+
+	for(var/obj/I in all_items)
+		if(istype(I, steal_target))

```

Comment:
this is a ```/datum/theft_objective```, so this needs to be ```if(istype(I, steal_target.typepath))```, now this is different to how /tg/ does it, and I've only just glanced at your code, but @Fox-McCloud confirm?